helix-tui/src/buffer.rs

```diff
@@ -433,6 +433,19 @@ impl Buffer {
         (x_offset as u16, y)
     }
 
+    pub fn set_spans_truncated(&mut self, x: u16, y: u16, spans: &Spans, width: u16) -> (u16, u16) {
+        let mut text = String::new();
+        let mut styles = vec![];
+
+        for span in &spans.0 {
+            text.push_str(span.content.as_ref());
+            span.styled_graphemes(span.style)
```

Comment:
Performing grapheme segmentation here is both slow and incorrect. The `idx` used below is not a grapheme index but a byte index. Instead you should create an iterator that is transversed whenever the current style range is exhausted.

The current implementation can easily lead to crashes

Reply:
I figured this was a slow solution.
Could you elaborate a bit about what you mean and how I could go about correcting this? I'm pretty new to this kind of thing.